src/core/src/sys/compute.rs

```diff
@@ -277,8 +286,11 @@ impl Compute for Stress {
             kinetic += particle.mass * velocity.tensorial(velocity);
         }
 
-        let virial = Virial.compute(system);
-        let volume = Volume.compute(system);
+        let volume = match system.cell().shape() {
```

Comment:
This piece of code appears three times, maybe it would be useful to put it in a method ?